extensions/amp-playbuzz/0.1/amp-playbuzz.js

```diff
@@ -0,0 +1,263 @@
+import {postMessage} from '../../../src/iframe-helper';
+
+/** @const */
+const EXPERIMENT = 'amp-playbuzz';
+
+class AmpPlaybuzz extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {?Element} */
+    this.iframe_ = null;
+
+    /** @private {?Promise} */
+    this.iframePromise_ = null;
+
+    /** @private {?string} */
+    this.item_ = '';
+
+     /** @private {?number} */
+    this.itemHeight_ = 300; //default
+
+     /** @private {?boolean} */
+    this.displayItemInfo_ = false;
+
+     /** @private {?boolean} */
+    this.displayShareBar_ = false;
+
+     /** @private {?boolean} */
+    this.displayComments_ = false;
+  }
+  /**
+   * @override
+   */
+  preconnectCallback() {
+    this.preconnect.preload(this.item_);
+  }
+
+  /** @override */
+  renderOutsideViewport() {
+    return false;
+  }
+
+  /** @override */
+  buildCallback() {
+    // EXPERIMENT
+    AMP.toggleExperiment(EXPERIMENT, true); //for dev
+    user().assert(isExperimentOn(this.win, EXPERIMENT),
+      `Enable ${EXPERIMENT} experiment`);
+
+    const e = this.element;
+    this.item_ = user().assert(
+      e.getAttribute('item-url'),
+      'The item attribute is required for <amp-playbuzz> %s',
+      e);
+    const parsedHeight = parseInt(e.getAttribute('height'), 10);
+    this.itemHeight_ = isNaN(parsedHeight) ? this.itemHeight_ : parsedHeight;
+    this.displayItemInfo_ = e.getAttribute('display-item-info') === 'true';
+    this.displayShareBar_ = e.getAttribute('display-share-buttons') === 'true';
+    this.displayComments_ = e.getAttribute('display-comments') === 'true';
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return layout === Layout.RESPONSIVE;
+    // return layout === Layout.CONTAINER;
+    // return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  createPlaceholderCallback() {
+    const placeholder = this.win.document.createElement('div');
+    setStyles(placeholder, {
+      'max-height': '300px',
+      'height': '300px',
+    });
+    placeholder.setAttribute('placeholder', '');
+    placeholder.appendChild(this.createPlaybuzzLoader_());
+    return placeholder;
+  }
+
+  /** @override */
+  layoutCallback() {
+    const iframe = this.element.ownerDocument.createElement('iframe');
+    this.iframe_ = iframe;
+    iframe.setAttribute('frameborder', '0');
+    iframe.setAttribute('allowtransparency', 'true');
+    iframe.src = this.generateEmbedSourceUrl_();
+
+    this.listenToPlaybuzzItemMessage_('resize_height',
+      this.setElementSize_.bind(this));
+
+    whenDocumentComplete(this.win.document)
+      .then(this.setElementSize_.bind(this));
+
+    this.applyFillContent(iframe);
+    this.element.appendChild(iframe);
+    setStyles(iframe, {
+      'opacity': 0,
+      'min-height': '300px',
+      'height': '300px',
+    });
+    setStyles(this.element, {'height': '300px', 'min-height': '300px'});
```

Comment:
It shouldn't be getting 100%, but possible if you CSS somewhere resizes it this way. For now, let's focus on user-specified height and let it do the work. We can look if we can/need do default later. In the example above, the responsive layout will automatically resize you embed to take all available width and then will set the height proportional to the width. In that particular case, given `layout=responsive width=300 height=300`, the `height` will be equal to the `width`. If that's not the case, please check your CSS and if you don't see the reason - let me know and I'll check it on my end.